server/konnectors/aprr.js

```diff
@@ -0,0 +1,202 @@
+  vendor: 'APRR',
+  dateFormat: 'YYYYMMDD',
+};
+
+const Bill = require('../models/bill');
+
+// Konnector
+const connector = module.exports = baseKonnector.createNew({
+  name: 'APRR',
+  fields: {
+    login: 'text',
+    password: 'password',
+    folderPath: 'folder',
+  },
+  models: [Bill],
+  fetchOperations: [
+    getHiddenInputs,
+    logIn,
+    fetchBillingInfo,
+    parsePage,
+    customFilterExisting,
+    customSaveDataAndFile,
+    linkBankOperation({
+      log,
+      model: Bill,
+      identifier: 'APRR',
+      minDateDelta: 4,
+      maxDateDelta: 20,
+      amountDelta: 0.1,
+    }),
+    buildNotifContent,
+  ],
+});
+
+// Procedure to get hidden inputs
+function getHiddenInputs(requiredFields, bills, data, next) {
+  const url = 'https://espaceclient.aprr.fr/aprr/Pages/connexion.aspx';
+  const options = {
+    url,
+    method: 'GET',
+  };
+
+  connector.logger.info('Getting the hidden inputs...');
+
+  request(options, (err, res, body) => {
+    if (err) return next(err);
+
+    const $ = cheerio.load(body);
+
+    var obj = {};
+    $( "body" ).find("input[type='hidden']").each(function(){
+		obj[$(this).attr('name')] = $(this).val();
+	});
+
+	// adding login/pwd
+	obj['ctl00$PlaceHolderMain$TextBoxLogin'] = requiredFields.login;
+	obj['ctl00$PlaceHolderMain$TextBoxPass'] = requiredFields.password;
+	obj['ctl00$PlaceHolderMain$ImageButtonConnection.x'] = Math.floor((Math.random() * 10) + 1);
+	obj['ctl00$PlaceHolderMain$ImageButtonConnection.y'] = Math.floor((Math.random() * 10) + 1);
+
+	data.inputs = obj;
+
+    //connector.logger.info('inputs', data.inputs);
+    return next();
+  });
+};
+
+
+// Procedure to login
+function logIn(requiredFields, bills, data, next) {
+
+  const options = {
+    method: 'POST',
+    url: 'https://espaceclient.aprr.fr/aprr/Pages/connexion.aspx',
+    form: data.inputs,
+	headers: {
+        'Referer': 'https://espaceclient.aprr.fr/aprr/Pages/connexion.aspx',
+     },
+  };
+
+  connector.logger.info('Logging in on APRR website...');
+
+  request(options, (err, res, body) => {
+    if (err) return next(err);
+
+    connector.logger.info('Successfully logged in.');
+
+    return next();
+  });
+};
+
+function fetchBillingInfo(requiredFields, bills, data, next) {
+  const url = 'https://espaceclient.aprr.fr/aprr/Pages/MaConsommation/conso_factures.aspx';
+
+  connector.logger.info('Fetch bill info');
+  const options = {
+    method: 'GET',
+    url,
+  };
+
+  request(options, (err, res, body) => {
+    if (err) {
+      log.error('An error occured while fetching bills');
+      log.raw(err);
+      return next(err);
+    }
+    connector.logger.info('Fetch bill info succeeded');
+
+    data.html = body;
+
+    return next();
+  });
+};
+
+function parsePage(requiredFields, bills, data, next) {
+  bills.fetched = [];
+  moment.locale('fr');
+  const $ = cheerio.load(data.html);
+  var billFoundNumber = 0;
+
+	$('table[class=tbl_factures] tr').each(function (i, tr) {
```

Comment:
hi,
i executed linting, and I have a question : which ES version should I be using ?
is cozy compatible with ES6 and for example templates (http://eslint.org/docs/rules/prefer-template) ?